generation/hermetic_library_generation.sh

```diff
@@ -75,9 +78,19 @@ message="chore: generate libraries at $(date)"
 
 git checkout "${target_branch}"
 git checkout "${current_branch}"
+# if the last commit doesn't contain changes to generation configuration,
+# do not generate again as the result will be the same.
+contains_config_change=$(git diff-tree --no-commit-id --name-only HEAD~1..HEAD -r | grep "${generation_config}")
+if [[ "${contains_config_change}" == "" ]]; then
+    echo "The last commit doesn't contain any changes to the generation_config.yaml, skipping the whole generation process."
+    exit 0
+fi
 # copy generation configuration from target branch to current branch.
 git show "${target_branch}":"${generation_config}" > "${baseline_generation_config}"
-diff "${generation_config}" "${baseline_generation_config}" || echo "config diff"
+config_diff=$(diff "${generation_config}" "${baseline_generation_config}")
+echo "Configuration diff:"
```

Comment:
Can we move this diff logging to right before `git commit` to prevent possible double logging?

Reply:
done.